lib/ReactViews/Map/Panels/SettingPanel.tsx

```diff
@@ -247,15 +253,17 @@ class SettingPanel extends React.Component<PropTypes> {
             <Text as="label">{t("settingPanel.mapView")}</Text>
           </Box>
           <FlexGrid gap={1} elementsNo={3}>
-            {Object.entries(MapViewers).map(([key, viewerMode]) => (
-              <SettingsButton
-                key={key}
-                isActive={key === currentViewer}
-                onClick={(event: any) => this.selectViewer(key as any, event)}
-              >
-                <Text mini>{t(viewerMode.label)}</Text>
-              </SettingsButton>
-            ))}
+            {Object.entries(MapViewers)
```

Comment:
The previous line contains `elementsNo={3}`, which coincides with `MapViewers.length`. Does this work as intended when the filtering filters out some objects?

And even without filtering, why does the previous line not need to be updated to contain 4 since there is one more item in MapViewers in this PR?